src/toplevel.rs

```diff
@@ -199,15 +199,18 @@ impl<ErrType: ErrTypeTraits> Toplevel<ErrType> {
             .filter_map(|exit_state| exit_state.raw_result.err())
             .collect::<Vec<_>>();
 
+        let mut log_message = String::new();
         // Print subsystem exit states
         if failed_subsystems.is_empty() {
-            log::debug!("Shutdown successful. Subsystem states:");
+            log_message.push_str("Shutdown successful.\n");
         } else {
-            log::debug!("Some subsystems failed. Subsystem states:");
+            log_message.push_str("Some subsystems failed.\n");
         };
+        log_message.push_str("Subsystem states:\n");
         for formatted_exit_state in formatted_exit_states {
-            log::debug!("    {}", formatted_exit_state);
+            log_message.push_str(format!("    {}\n", formatted_exit_state).as_str());
         }
+        tracing::debug!("{}", log_message);
```

Comment:
Leave as-is. After doing some research it seems that multi-line logs are quite discouraged, and multiple single-line logs are preferred.